test-context/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+use postgresql_embedded::{PostgreSQL, Settings, VersionReq};
+use std::convert::Infallible;
+use std::env;
+use std::env::current_dir;
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use test_context::futures::stream;
+use test_context::AsyncTestContext;
+use tokio::io::AsyncReadExt;
+use tokio_util::bytes::Bytes;
+use tracing::{info_span, instrument, Instrument};
+use trustify_common as common;
+use trustify_module_ingestor::graph::Graph;
+use trustify_module_ingestor::service::{Format, IngestorService};
+use trustify_module_storage::service::fs::FileSystemBackend;
+
+#[allow(dead_code)]
+pub struct TrustifyContext {
+    pub db: common::db::Database,
+    pub graph: Graph,
+    pub storage: FileSystemBackend,
+    pub ingestor: IngestorService,
+    postgresql: Option<PostgreSQL>,
+}
+
+impl TrustifyContext {
+    pub async fn ingest_documents<'a, P: IntoIterator<Item = &'a str>>(
+        &self,
+        paths: P,
+    ) -> Result<(), anyhow::Error> {
+        let workspace_root = find_workspace_root()?;
+        let test_data = workspace_root.join("etc").join("test-data");
+
+        for path in paths {
```

Comment:
What would be handy (for the sbom test I asked you to change, in particular) would be to return a mapping of path to `IngestResult`.

Reply:
Yep, done!

Added results when multi-ingest (indexed by input path), and also a single simple route for a single-doc ingest.